### PR TITLE
Fix analyze iceberg table with an uppercase column name

### LINKED ISSUE #14583: When the column name of the Iceberg metadata file record is an uppercase field, executing ANALYZE TABLE will report that the column cannot be found.
Flink SQL create iceberg table with an uppercase column name:
```
create table b (B int) ;
```
and Trino which version I used is v397 executes ANALYZE b,met this error.

```
Query 20221012_053136_00006_fw48v failed: Column not found in table: [b]
com.google.common.base.VerifyException: Column not found in table: [b]
	at com.google.common.base.Verify.verifyNotNull(Verify.java:503)
	at io.trino.plugin.iceberg.IcebergMetadata.finishStatisticsCollection(IcebergMetadata.java:1538)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.finishStatisticsCollection(ClassLoaderSafeConnectorMetadata.java:166)
	at io.trino.metadata.MetadataManager.finishStatisticsCollection(MetadataManager.java:812)
	at io.trino.sql.planner.LocalExecutionPlanner$Visitor.lambda$visitStatisticsWriterNode$44(LocalExecutionPlanner.java:3218)
	at io.trino.operator.StatisticsWriterOperator.getOutput(StatisticsWriterOperator.java:133)
	at io.trino.operator.Driver.processInternal(Driver.java:411)
	at io.trino.operator.Driver.lambda$process$10(Driver.java:314)
	at io.trino.operator.Driver.tryWithLock(Driver.java:706)
	at io.trino.operator.Driver.process(Driver.java:306)
	at io.trino.operator.Driver.processForDuration(Driver.java:277)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:736)
	at io.trino.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:164)
	at io.trino.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:515)
	at io.trino.$gen.Trino_397____20221011_085503_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

### PR DESCRIPTION
**Pull request by zhengxingmao**
## Description
Fix #14583

## Release notes

(*) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`14583`)
```

### CODE REVIEW

**Comment by findepi** [comment]
@zhengxingmao thank you for your PR!

Would it be possible to have a test for this?

You will need Spark for creating a table with upper-case fields, you can add a test case to `TestIcebergSparkCompatibility` test class.

**Comment by zhengxingmao** [comment]
> @zhengxingmao thank you for your PR!
> 
> Would it be possible to have a test for this?
> 
> You will need Spark for creating a table with upper-case fields, you can add a test case to `TestIcebergSparkCompatibility` test class.

**Comment by zhengxingmao** [comment]
> @zhengxingmao thank you for your PR!
> 
> Would it be possible to have a test for this?
> 
> You will need Spark for creating a table with upper-case fields, you can add a test case to `TestIcebergSparkCompatibility` test class.

Done.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java [hunk 1]
```diff
@@ -2183,6 +2183,25 @@ public void testTrinoAnalyze()
         onTrino().executeQuery("DROP TABLE " + trinoTableName);
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testTrinoAnalyzeWithUppercaseFiled()
```
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java:2187]: Filed -> Field

**Review by findepi — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java (hunk 1 above)
Comment (findepi) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java]: pushed a fix; changed to `testTrinoAnalyzeWithNonLowercaseColumnName`

**Pull request merged**